prisma, prisma-engines: 6.19.0 -> 7.0.1

jnsgruk opened the pull request:
This PR updates `prisma` and `prisma-engines` from `6.19.0` -> `7.0.1`.

Because this is a breaking change, and some packages have yet to migrate to Prisma 7, this PR does the following:

- Renames the existing `prisma` -> `prisma_6`
- Renames the existing `prisma-engines` -> `prisma-engines_6`
- Introduces `prisma_7` and `prisma-engines_7`
- Creates an alias `prisma` -> `prisma_7`
- Creates an alias `prisma-engines` -> `prisma-engines_7`

As a drive-by, `prisma_6` is also updated from `6.19.0` -> `6.19.1`.

This is a follow-on from #467188, with thanks to @mahmoudajawad for the original work on Prisma 7.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jnsgruk committed:
treewide: rename `prisma` -> `prisma_6`, `prisma-engines` -> `prisma-engines_6`

jnsgruk committed:
prisma, prisma-engines: 6.19.0 -> 7.0.1

jnsgruk committed:
prisma_6, prisma-engines_6: 6.19.0 -> 6.19.1

aqrln commented on pkgs/by-name/pr/prisma-engines_7/setup-hook.sh:
```diff
@@ -0,0 +1,2 @@
+export PRISMA_SCHEMA_ENGINE_BINARY="@out@/bin/schema-engine"
+export PRISMA_FMT_BINARY="@out@/bin/prisma-fmt"
```
```suggestion
```

jnsgruk replied:
Fixed in https://github.com/NixOS/nixpkgs/pull/472380/commits/91ad6aff95484cffede2dd90b7f2a02e0099e05f

jnsgruk committed:
prisma-engines_6, prisma-engines_7: remove unused `protobuf`

jnsgruk committed:
prisma-engines_6, prisma-engines_7: remove `tomhoule` from maintainers

jnsgruk committed:
prisma-engines_7: remove `PRISMA_FMT_BINARY` env var from `setup-hook.sh`

aqrln commented on pkgs/by-name/pr/prisma-engines_6/package.nix:
```diff
@@ -12,14 +11,14 @@
+    hash = "sha256-z3GdnrLEMJIGPKXXbz2wrbiGpuNlgYxqg3iYINYTnPI=";
   };
 
   cargoHash = "sha256-PgCfBcmK9RCA5BMacJ5oYEpo2DnBKx2xPbdLb79yCCY=";
```
This probably needs to be updated too, I'm seeing version 6.19.0 for vendored dependencies in [build logs](https://github.com/aqrln/nixpkgs-review-gha/actions/runs/20374098615/job/58547902949) since it's a FOD:

```
...
prisma-engines_6> Unpacking to /nix/store/0qjydpxqy50vpv5mdnx9d7zalz4hj2xf-prisma-engines_6-6.19.0-vendor/darling-0.20.11
prisma-engines_6> Unpacking to /nix/store/0qjydpxqy50vpv5mdnx9d7zalz4hj2xf-prisma-engines_6-6.19.0-vendor/darling_core-0.20.11
prisma_6-pnpm-deps> Progress: resolved 1491, reused 0, downloaded 1480, added 489
prisma_7-pnpm-deps> Progress: resolved 1557, reused 0, downloaded 1445, added 408
prisma-engines_6> Unpacking to /nix/store/0qjydpxqy50vpv5mdnx9d7zalz4hj2xf-prisma-engines_6-6.19.0-vendor/darling_macro-0.20.11
prisma_6-pnpm-deps> Progress: resolved 1491, reused 0, downloaded 1480, added 684
prisma-engines_6> Unpacking to /nix/store/0qjydpxqy50vpv5mdnx9d7zalz4hj2xf-prisma-engines_6-6.19.0-vendor/dashmap-5.5.0
prisma-engines_6> Unpacking to /nix/store/0qjydpxqy50vpv5mdnx9d7zalz4hj2xf-prisma-engines_6-6.19.0-vendor/data-encoding-2.4.0
prisma-engines_6> Unpacking to /nix/store/0qjydpxqy50vpv5mdnx9d7zalz4hj2xf-prisma-engines_6-6.19.0-vendor/deranged-0.3.11
prisma_6-pnpm-deps> Progress: resolved 1491, reused 0, downloaded 1480, added 503
prisma-engines_6> Unpacking to /nix/store/0qjydpxqy50vpv5mdnx9d7zalz4hj2xf-prisma-engines_6-6.19.0-vendor/derive-syn-parse-0.2.0
prisma-engines_6> Unpacking to /nix/store/0qjydpxqy50vpv5mdnx9d7zalz4hj2xf-prisma-engines_6-6.19.0-vendor/derive-where-1.6.0
prisma_7-pnpm-deps> Progress: resolved 1557, reused 0, downloaded 1450, added 408
prisma-engines_6> Unpacking to /nix/store/0qjydpxqy50vpv5mdnx9d7zalz4hj2xf-prisma-engines_6-6.19.0-vendor/derive_more-0.99.20
...
```

jnsgruk committed:
prisma_7: specify `pnpm_10` in `fetchPnpmDeps` block

diogotcorreia commented on pkgs/by-name/pr/prisma-engines_6/package.nix:
```diff
@@ -12,14 +11,14 @@
   src = fetchFromGitHub {
     owner = "prisma";
     repo = "prisma-engines";
     rev = version;
```
```suggestion
    tag = version;
```

Nit, but since we're changing a bunch of stuff might as well.

Ditto for all other packages (and if you don't mind, do the same in the prisma overrides in the umami package)

diogotcorreia reviewed (commented):
Umami diff LGTM

jnsgruk committed:
treewide: `s/rev/tag/g` on prisma source blocks

Moraxyc commented on pkgs/by-name/gh/ghostfolio/package.nix:
ghostfolio diff LGTM, thank you!

diogotcorreia reviewed (approved):
Ran umami tests, everything works, thanks!

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 472380 --additional-package umami.tests`
Commit: `312848e061354d892094e1ca564e4e49738c17fd`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>documenso</li>
    <li>ghostfolio</li>
    <li>linkwarden</li>
    <li>prisma-engines_6</li>
    <li>prisma-engines_7</li>
    <li>prisma_6</li>
    <li>prisma_7</li>
    <li>umami</li>
    <li>umami.tests.umami</li>
    <li>zipline</li>
  </ul>
</details>

jnsgruk committed:
prisma-engines_7: don't build `prisma-fmt`

jnsgruk commented on pkgs/by-name/pr/prisma-engines_7/package.nix:
```diff
@@ -0,0 +1,76 @@
+    "-p"
+    "schema-engine-cli"
+    "-p"
+    "prisma-fmt"
```
Fixed in https://github.com/NixOS/nixpkgs/pull/472380/commits/688c34105ce3ae93675b898f725f8d52d1124bdf

happysalada reviewed (approved):
Diff looks good

jnsgruk committed:
prisma_6: drop `prisma init` in test which now requires internet access

Later versions of Prisma 6 seem to require internet
access on the first run of `prisma init` to fetch
updates to the subcommand.

This fails in the test environment, but since we
are already sketching out the test schema structure
manually, we can drop that in favour of just creating
the prisma/ directory.

aqrln reviewed (approved):
LGTM!

diogotcorreia commented:
Does it make sense to backport this, but keep the alias as `prisma{,-engines}` -> `prisma{,-engines}_6`?
At some point apps are going to update to require Prisma 7 and it's also useful for local development.

If no one opposes I don't mind opening a PR soon.
